lib/jekyll/filters.rb

```diff
@@ -221,7 +221,9 @@ def where(input, property, value)
       return input unless input.respond_to?(:select)
       input = input.values if input.is_a?(Hash)
       input.select do |object|
-        Array(item_property(object, property)).map(&:to_s).include?(value.to_s)
+        item_value = item_property(object, property)
+        item_value = [item_value] unless item_value.respond_to?("map")
+        item_value.map(&:to_s).include?(value.to_s)
```

Comment:
The problem with special-casing nil is that it would break the existing behavior. `where` has always treated nil and empty-string (as well as nonexistent properties) as identical because they all stringify to the empty-string. The array-support added by #4555 continued this tradition.

I've added some more tests to make the current handling of nil/empty-string in list properties explicit. A special case for nil will break these tests.